### PR TITLE
Vehicle: keep commands in list if in progress

### PR DESCRIPTION
**Pull request by julianoes**
We should not remove commands from the command list when we receive an IN_PROGRESS ack because we should eventually receive a final ack.

### CODE REVIEW

**Review by patrickelectric — APPROVED** [review]

**Comment by peterbarker** [comment]
Ah, glad you're doing this; I was looking at changing the ArduPilot-specific calibration stuff to use this mechanism....

File: src/Vehicle/Vehicle.cc [hunk 1]
```diff
@@ -3254,7 +3254,10 @@ void Vehicle::_handleCommandAck(mavlink_message_t& message)
     int entryIndex = _findMavCommandListEntryIndex(message.compid, static_cast<MAV_CMD>(ack.command));
     bool commandInList = false;
     if (entryIndex != -1) {
-        MavCommandListEntry_t commandEntry = _mavCommandList.takeAt(entryIndex);
+        // We remove the command from the list unless it's still in progress.
```
Comment (bkueng) [inline src/Vehicle/Vehicle.cc:3257]: While in progress, are retries handled correctly?

**Review by bkueng — COMMENTED** [review]
The change is fine with me, but [autotune](https://github.com/mavlink/qgroundcontrol/blob/master/src/Vehicle/Autotune.cpp) does not follow that semantics and needs to be checked.

**Comment by DonLakeFlyer** [comment]
Beat is correct. Supporting this is more involved than just the list management. I'll take a look.

**Comment by DonLakeFlyer** [comment]
There are also big problems with how MAV_RESULT_IN_PROGRESS and progress values were plumbed through Vehicle::sendMavCommandWithHandler. Given the fact that any command can start to send IN_PROGRESS before sending an ack, all callbacks from Vehicle::sendMavCommandWithHandler as currently implemented need to handle IN_PROGRESS specifically otherwise they will break if the command ever sends that.

**Comment by DonLakeFlyer** [comment]
I should have a pull which cleans this all up in a day or so...

**Comment by julianoes** [comment]
Closing this. Thanks @DonLakeFlyer for #10852.

**Pull request closed without merging**